tests/unit/test_charm.py

```diff
@@ -0,0 +1,533 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import io
+import logging
+from pathlib import Path
+from unittest.mock import patch
+
+import ops.testing
+import pytest
+import yaml
+from ops.framework import EventBase
+from ops.model import ActiveStatus, MaintenanceStatus, WaitingStatus
+from ops.testing import Harness
+
+from charm import ZooKeeperK8sCharm
+from literals import CHARM_KEY, CONTAINER, PEER
+
+ops.testing.SIMULATE_CAN_CONNECT = True
+logger = logging.getLogger(__name__)
+
+CONFIG = str(yaml.safe_load(Path("./config.yaml").read_text()))
+ACTIONS = str(yaml.safe_load(Path("./actions.yaml").read_text()))
+METADATA = str(yaml.safe_load(Path("./metadata.yaml").read_text()))
+
+
+@pytest.fixture
+def harness():
```

Comment:
Should you just add `peer_rel_id = harness.add_relation(PEER, CHARM_KEY)` here instead?

Reply:
I wanted to test the scenario where `if not self.cluster.relation' somwhere I think.